src/Mvc/Extensions.ApiDescription.Client/src/buildMultiTargeting/Microsoft.Extensions.ApiDescription.Client.targets

```diff
@@ -1,9 +1,14 @@
 ﻿<?xml version="1.0" encoding="utf-8" standalone="no"?>
 <Project>
-  <Target Name="GenerateServiceFileReferenceCode" BeforeTargets="BeforeCompile">
-    <MsBuild Projects="$(MSBuildProjectFile)"
-        Targets="GenerateServiceFileReferenceCode"
+  <Target Name="GenerateOpenApiReferenceCode">
+    <MSBuild Projects="$(MSBuildProjectFile)"
+        Targets="GenerateOpenApiReferenceCode"
         Properties="TargetFramework=$(TargetFrameworks.Split(';')[0])"
-        RemoveProperties="TargetFrameworks;RuntimeIdentifier" />
+        RemoveProperties="RuntimeIdentifier" />
   </Target>
+
+  <Target Name="_TieInGenerateOpenApiReferenceCode"
+      BeforeTargets="BeforeCompile"
```

Comment:
Should this be `CoreCompile`. Would that allow users to do user things ™️ in `BeforeCompile`?

Reply:
`CoreCompile` is late because it comes after the MSBuild targets cache the list of references and source files involved in the compilation